t/app/controller/waste_peterborough.t

```diff
@@ -821,9 +822,137 @@ FixMyStreet::override_config {
         };
-
     };
 
+    subtest 'WasteWorks bulky goods item list administration' => sub {
+        ok $mech->host('peterborough.fixmystreet.com');
+        my ($b, $jobs_fsd_get) = shared_bartec_mocks();
+
+        subtest 'List admin page is linked from config page' => sub {
+            $mech->log_in_ok($super->email);
+            $mech->get_ok('/admin/waste/' . $body->id);
+            $mech->follow_link_ok( { text_regex => qr/Bulky items list/i, }, "follow 'Bulky items list' link" );
+            is $mech->uri->path, '/admin/waste/' . $body->id . '/bulky_items', 'ended up on correct page';
+        };
+
+        subtest 'Items can be stored correctly' => sub {
+            $body->set_extra_metadata(wasteworks_config => {});
+            $body->update;
+
+            $mech->get_ok('/admin/waste/' . $body->id . '/bulky_items');
+            $mech->submit_form_ok({ with_fields => {
+                'bartec_id[9999]' => 1234,
+                'category[9999]' => 'Furniture',
+                'name[9999]' => 'Sofa',
+                'price[9999]' => '0',
+                'message[9999]' => 'test',
+            }});
+            $mech->content_contains("Updated!");
+
+            $body->discard_changes;
+            is_deeply $body->get_extra_metadata('wasteworks_config'), {
+                item_list => [ {
+                    bartec_id => "1234",
+                    category => "Furniture",
+                    message => "test",
+                    name => "Sofa",
+                    price => "0"
+                }]
+            };
+
+            $mech->submit_form_ok({ with_fields => {
+                'bartec_id[9999]' => 4567,
+                'category[9999]' => 'Furniture',
+                'name[9999]' => 'Armchair',
+                'price[9999]' => '10',
+                'message[9999]' => '',
+            }});
+
+            $body->discard_changes;
+            is_deeply $body->get_extra_metadata('wasteworks_config'), {
+                item_list => [
+                    {
+                        bartec_id => "4567",
+                        category => "Furniture",
+                        message => "",
+                        name => "Armchair",
+                        price => "10"
+                    },
+                    {
+                        bartec_id => "1234",
+                        category => "Furniture",
+                        message => "test",
+                        name => "Sofa",
+                        price => "0"
+                    },
+                ]
+            };
+        };
+
+        subtest 'Bartec feature list is shown correctly' => sub {
+            $body->set_extra_metadata(wasteworks_config => {});
+            $body->update;
+
+            $b->mock('Features_Types_Get', sub { [
+                {
+                    Name => "Bookcase",
+                    ID => 6941,
+                    FeatureClass => {
```

Comment:
Is FeatureClass something like a category? Is there a way to convert the ID to a category string?

Reply:
Not really - it's either `BULKY` or `BULKY WHITES`, I believe.